src/main/java/duke/Ui.java

```diff
@@ -66,20 +102,39 @@ public void showTaskList(TaskList tasks) throws DukeException {
         }
     }
 
+    /**
+     * Prints a message to acknowledge the addition of the task to the TaskList.
+     * @param task The task added to the TaskList.
+     * @param tasks The TaskList where the task is added.
+     */
     public void showAddTask(Task task, TaskList tasks) {
         showToUser("Got it. I've added this task:", "  " + task, 
                 "Now you have " + tasks.size() + " tasks in the list.");
     }
 
-    public void showDoneTask(Task task, TaskList tasks) {
+    /**
+     * Prints a message to acknowledge that the task has been marked as done.
+     * @param task The task marked as done.
+     */
+    public void showDoneTask(Task task) {
         showToUser("Nice! I've marked this task as done: ", "  " + task);
     }
 
+    /**
+     * Prints a message to acknowledge that the task has been delete from the TaskList.
+     * @param task The task to be deleted from the TaskList.
+     * @param tasks The TaskList where the task is deleted.
+     */
     public void showDeleteTask(Task task, TaskList tasks) {
         showToUser("Noted. I've removed this task: ", "  " + task, 
                 "Now you have " + tasks.size() + " tasks in the list.");
     }
```

Comment:
I think this is a good use of abstraction. For me, I only had a printMessage method to cater for these messages. It's definitely something I'll consider integrating into my project. Good job!